python/semantic_kernel/connectors/memory/qdrant/qdrant_memory_store.py

```diff
@@ -0,0 +1,385 @@
+            points=[keys],
+            ),
+        )
+
+    async def get_nearest_match_async(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        min_relevance_score: float = 0.0,
+        with_embedding: bool = False,
+    ) -> Tuple[MemoryRecord, float]:
+        """Gets the nearest match to an embedding using cosine similarity.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to get the nearest match from.
+            embedding {ndarray} -- The embedding to find the nearest match to.
+            min_relevance_score {float} -- The minimum relevance score of the match. (default: {0.0})
+            with_embedding {bool} -- Whether to include the embedding in the result. (default: {False})
+
+        Returns:
+            Tuple[MemoryRecord, float] -- The record and the relevance score.
+        """
+        return self.get_nearest_matches_async(
+            collection_name=collection_name,
+            embedding=embedding,
+            limit=1,
+            min_relevance_score=min_relevance_score,
+            with_embeddings=with_embedding,
+        )
+
+    async def get_nearest_matches_async(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        limit: int,
+        min_relevance_score: float = 0.0,
+        with_embeddings: bool = False,
+    ) -> List[Tuple[MemoryRecord, float]]:
+        """Gets the nearest matches to an embedding using cosine similarity.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to get the nearest matches from.
+            embedding {ndarray} -- The embedding to find the nearest matches to.
+            limit {int} -- The maximum number of matches to return.
+            min_relevance_score {float} -- The minimum relevance score of the matches. (default: {0.0})
+            with_embeddings {bool} -- Whether to include the embeddings in the results. (default: {False})
+
+        Returns:
+            List[Tuple[MemoryRecord, float]] -- The records and their relevance scores.
+        """
+        
+        collection_info = self._qdrantclient.get_collection(collection_name=collection_name)
+
+        if not collection_info.status == CollectionStatus.GREEN:
+            raise Exception(f"Collection '{collection_name}' does not exist")
+
+        # Search for the nearest matches, qdrant already provides results sorted by relevance score
+        matches = self._qdrantclient.search(
+            collection_name=collection_name,
+            search_params=models.SearchParams(
+                hnsw_ef=0,
+                exact=False,
+                quantization=None,
+            ),
+            query_vector=embedding,
+            limit=limit,
+            score_threshold=min_relevance_score,
+            with_vectors=with_embeddings,
+            with_payload=True,
+        )
+
+        # Convert the results to MemoryRecords
+
+    def compute_similarity_scores(
```

Comment:
I don't think this is being used

Reply:
Yeah I think we can drop this. Qdrant does this themselves right?

Reply:
Yes Qdrant does this as a part of their service.

Reply:
Moved to qdrant_utils.py in commit: https://github.com/microsoft/semantic-kernel/pull/919/commits/a0732ee615e58792189d17ed7e192908f61c7eee

Reply:
Please advise if the move for the function works.

Reply:
Addressed in PR https://github.com/microsoft/semantic-kernel/pull/919/commits/2d79e60727ea2dc9dd267bb35b298c069c32b5f3